src/overlays/actors/ovl_En_Osn/z_en_osn.c

```diff
@@ -74,59 +101,824 @@ static DamageTable D_80AD2550 = {
+            this->actionFunc = EnOsn_Idle;
+            break;
+
+        case 3:
+            this->actor.flags &= ~ACTOR_FLAG_1;
+            this->actionFunc = func_80AD16A8;
+            break;
+
+        default:
+            Actor_MarkForDeath(&this->actor);
+    }
+}
+
+void EnOsn_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnOsn* this = THIS;
+
+    SkelAnime_Free(&this->skelAnime, globalCtx);
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void EnOsn_Update(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnOsn* this = THIS;
+    u32 sp34;
+
+    sp34 = Flags_GetSwitch(globalCtx, 0);
+    this->actionFunc(this, globalCtx);
+    Actor_MoveWithGravity(&this->actor);
+    SkelAnime_Update(&this->skelAnime);
+    if (!(ENOSN_GET_3(&this->actor))) {
+        if (sp34 != 0) {
+            this->actor.flags |= ACTOR_FLAG_1;
+            func_80AD0830(this, globalCtx);
+            this->actor.draw = EnOsn_Draw;
+        } else {
+            this->actor.draw = NULL;
+            this->actor.flags &= ~ACTOR_FLAG_1;
+        }
+    }
+    func_800E9250(globalCtx, &this->actor, &this->unk_1D8, &this->unk_1DE, this->actor.focus.pos);
+}
+
+s32 EnOsn_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx,
+                           Gfx** gfx) {
+    EnOsn* this = (EnOsn*)thisx;
+
+    if (this->unk_1F0 && limbIndex == 11) {
+        Matrix_InsertXRotation_s(this->unk_1D8.y, MTXMODE_APPLY);
+    }
+    if ((this->unk_1EC == 9 || this->unk_1EC == 8) && limbIndex == 10) {
+        *dList = NULL;
+    }
+    return false;
+}
+
+void EnOsn_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* actor, Gfx** gfx) {
+    EnOsn* this = (EnOsn*)actor;
+    Vec3f sp30 = { 0.0f, 0.0f, 0.0f };
+    Vec3s sp28 = { 0x9920, -0x384, -0x320 };
+
+    if (limbIndex == 0xB) {
+        Matrix_MultiplyVector3fByState(&sp30, &actor->focus.pos);
+    }
+    if (((this->unk_1EC == 17) || (this->unk_1EC == 21) || (this->unk_1EC == 22)) && (limbIndex == 6)) {
+        Matrix_StatePush();
+        Matrix_InsertTranslation(-400.0f, 1100.0f, -200.0f, MTXMODE_APPLY);
+        Matrix_InsertXRotation_s(sp28.x, MTXMODE_APPLY);
+        Matrix_RotateY(sp28.y, MTXMODE_APPLY);
+        Matrix_InsertZRotation_s(sp28.z, MTXMODE_APPLY);
+
+        gSPMatrix((*gfx)++, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPDisplayList((*gfx)++, &object_osn_DL_0192A0);
+        Matrix_StatePop();
+    }
+}
+
+void EnOsn_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnOsn* this = THIS;
+    static TexturePtr D_80AD2588 = object_osn_Tex_0166F8;
+    static TexturePtr D_80AD258C = object_osn_Tex_016EF8;
+    static TexturePtr D_80AD2590 = object_osn_Tex_0176F8;
+    static TexturePtr D_80AD2594 = object_osn_Tex_017EF8;
+    static TexturePtr D_80AD2598 = object_osn_Tex_0182F8;
```

Comment:
```suggestion
    static TexturePtr D_80AD2588 = object_osn_Tex_0166F8;
    static TexturePtr D_80AD258C = object_osn_Tex_016EF8;
    static TexturePtr D_80AD2590 = object_osn_Tex_0176F8;
    static TexturePtr D_80AD2594 = object_osn_Tex_017EF8;
    static TexturePtr D_80AD2598 = object_osn_Tex_0182F8;
    s32 pad;
    EnOsn* this = THIS;
```
Minor, but generally we declare in function static variables first, then everything else.